### DISCUSSION TITLE
[Help] Retrieval data from Knowledge Dify

### DISCUSSION THREAD

**Discussion by olbboy** [langgenius/dify#6650] [Help]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

I added CSV data to the knowledge base, used the model text-embedding-ada-002 for embeddings, and employed the gpt-4o-mini model to generate responses for the chatbot. However, I encountered an issue with accurately retrieving the full quantity of information from the CSV file. For instance, when I asked for the total number of employees in department A, the response only provided data for 5 out of 15 employees. Upon checking the retrieval module, it supports a maximum topK of 10. How can I improve the chatbot's output for queries involving statistical counts?

**Comment by iamjoel** [comment]
Add some segments include statistical counts.

**Reply by olbboy** [reply]
@iamjoel Also an good idea, I came up with an idea that instead of depending on Dify's chunking and embedding capabilities, we can preprocess the data, transform it into a dataset and then put it into knowledge. Doing so will take time, the data size will be larger, but in return, the data retrieval will be more complete.
